site/public/js/server.js

```diff
@@ -1576,6 +1576,28 @@ function checkSidebarCollapse() {
     }
 }
 
+//Changes the theme from light to dark mode or the reverse
+//if mode='black' it will toggle the black mode instead of the normal mode
+function toggleTheme(mode='normal'){
+  if(mode==='normal'){
+    if((!localStorage.getItem("theme") && document.documentElement.getAttribute("data-theme") != "dark") || localStorage.getItem("theme") == "light"){
+        localStorage.setItem("theme", "dark");
+        document.documentElement.setAttribute("data-theme", "dark");
+    }else{
+      localStorage.setItem("theme", "light");
+      document.documentElement.setAttribute("data-theme", "light");
+    }
+  }else if(mode === 'black'){
+    if(!localStorage.getItem("black_mode") || localStorage.getItem("black_mode") != "black"){
```

Comment:
Can just do `localStorage.getItem('black_mode') !== 'black'` for test, as this covers both falsy values and incorrect string values.